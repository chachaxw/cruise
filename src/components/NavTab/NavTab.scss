@import '../../styles/_variables';

.cruise-nav-tab {
    display: flex;

    .nav-tab-item {
        height: 50px;
        line-height: 50px;
        padding: 0 25px;
        min-width: 80px;
        cursor: pointer;
        text-align: center;
        box-sizing: border-box;
        text-transform: capitalize;
        border-right: 1px solid #F3F3F3;
        border-bottom: 3px solid #FFFFFF;
        background-color: #FFFFFF;

        &.active {
            color: $cyanBlue;
            border-bottom-color: $cyanBlue;
        }
    }
}