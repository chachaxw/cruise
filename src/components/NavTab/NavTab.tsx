import * as React from 'react';
import './NavTab.scss';

export default class NavTab extends React.Component {
    
    public render() {
        return (
            <div className="nav-tab">
                Nav Tab
            </div>
        );
    }
}
