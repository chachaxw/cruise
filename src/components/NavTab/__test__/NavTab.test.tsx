import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import NavTab from '../NavTab';

it('renders correctly with given prop active', () => {
    const tabList = [
        { label: 'Test1', value: 0 },
        { label: 'Test2', value: 1 },
        { label: 'Test3', value: 2 },
    ];

    const testRenderer = TestRenderer.create(
        <NavTab active={0} tabList={[]} onTabChange={jest.fn()}></NavTab>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('renders correctly with given prop tabList', () => {
    const tabList = [
        { label: 'Test1', value: 0 },
        { label: 'Test2', value: 1 },
        { label: 'Test3', value: 2 },
    ];

    const testRenderer = TestRenderer.create(
        <NavTab active={0} tabList={tabList} onTabChange={jest.fn()}></NavTab>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('renders different active tab by click tab item', () => {
    const tabList = [
        { label: 'Test1', value: 0 },
        { label: 'Test2', value: 1 },
        { label: 'Test3', value: 2 },
    ];
    const testRenderer = TestRenderer.create(
        <NavTab active={0} tabList={tabList} onTabChange={jest.fn()}></NavTab>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
});