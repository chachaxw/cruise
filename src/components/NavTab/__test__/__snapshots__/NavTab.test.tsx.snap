// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with given prop active 1`] = `null`;

exports[`renders correctly with given prop tabList 1`] = `
<div
  className="cruise-nav-tab"
>
  <div
    className="nav-tab-item active"
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test1
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test2
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test3
    </span>
  </div>
</div>
`;

exports[`renders correctly with given prop tabList 2`] = `
<div
  className="cruise-nav-tab"
>
  <div
    className="nav-tab-item active"
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test1
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test2
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test3
    </span>
  </div>
</div>
`;

exports[`renders different active tab by click tab item 1`] = `
<div
  className="cruise-nav-tab"
>
  <div
    className="nav-tab-item active"
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test1
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test2
    </span>
  </div>
  <div
    className="nav-tab-item "
    onClick={[Function]}
  >
    <span
      className="nav-tab-item-label"
    >
      Test3
    </span>
  </div>
</div>
`;
